From: Liam Howlett <liam.howlett@oracle.com>
Subject: test_maple_tree: fix accounting in check_erase2_testset()
Date: Tue, 12 Jul 2022 14:53:11 +0000

When overwriting an empty area, do not decrement the number of expected
entries.

Link: https://lkml.kernel.org/r/20220712145303.4191493-1-Liam.Howlett@oracle.com
Fixes: 51282228cdd4 (lib/test_maple_tree: add testing for maple tree)
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/test_maple_tree.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/lib/test_maple_tree.c~lib-test_maple_tree-add-testing-for-maple-tree-fix-3
+++ a/lib/test_maple_tree.c
@@ -1486,7 +1486,8 @@ static noinline void check_erase2_testse
 		switch (set[i]) {
 		case SNULL:
 			if ((s_min == set[i+1]) && (s_max == set[i+2])) {
-				entry_count--;
+				if (s_entry)
+					entry_count--;
 			} else if ((s_min != set[i+1]) && (s_max != set[i+2])) {
 				entry_count++;
 			} else if ((mas_start.node != mas_end.node) ||
_
